lib/src/utils/type_checker.dart

```diff
@@ -124,14 +124,20 @@ const undoHistoryControllerChecker = TypeChecker.fromName(
   packageName: 'flutter',
 );
 
+const changeNotifierChecker = TypeChecker.fromName(
+  'ChangeNotifier',
+  packageName: 'flutter',
+);
+
 const disposableControllerChecker = TypeChecker.any(
   [
+    // Only the AnimationController does not extends ChangeNotifier class
     animationControllerChecker,
-    pageControllerChecker,
-    scrollControllerChecker,
-    searchControllerChecker,
-    tabControllerChecker,
-    textEditingControllerChecker,
-    undoHistoryControllerChecker,
+    // pageControllerChecker,
+    // scrollControllerChecker,
+    // searchControllerChecker,
+    // tabControllerChecker,
+    // textEditingControllerChecker,
+    // undoHistoryControllerChecker,
   ],
```

Comment:
Not sure if we want to keep `disposableControllerChecker` if there is only `animationControllerChecker`.

I guess we could keep it for now in case there are other built-in disposable controllers that aren't extends `ChangeNotifier`.

But we should definitely remove the commented lines.

Reply:
Sure